CONTRIBUTING.md

```diff
@@ -45,6 +59,13 @@ You can view them with:
 $ git status
 ```
 
+If you are changing anything with files and folder structure, 
+remember to update doctree
+```sh
+$ cd docs
+$ sphinx-apidoc -f -P -F -o source ../napari
```

Comment:
Should this instruction not just be part of the build process? My guide for what needs to be committed in git is that *anything generated* should not be committed. I think @royerloic was running into this problem yesterday after rebasing his dimensions branch.

Reply:
@jni It depends. This is only one way of generating doc source from the docstrings. IMO, sphinx commands can easily result in different ways, so enforcing single way of generating dev-docs sounds good to me for consistency. We can try and see if this brings any problems, also love to see what problem @royerloic had. 

Also, if we do not want to have people update doc source with their PRs, we need to have doc source update PRs regularly, which could be more annoying. (PS: I would not like to have any generated files too, with more reliable tools/packages/environments.)

Reply:
But this is the opposite of a single way of generating dev docs: with this system, they will be generated by any PR author (who touches the package structure) with their own system.